src/rapid_pro_to_engagement_db/rapid_pro_to_engagement_db.py

```diff
@@ -191,8 +201,19 @@ def sync_rapid_pro_to_engagement_db(rapid_pro, engagement_db, uuid_table, rapid_
             # De-identify the contact's full urn.
             contact = contacts_lut[run.contact.uuid]
             assert len(contact.urns) == 1, len(contact.urns)
-            contact_urn = contact.urns[0]
-            participant_uuid = _de_identify_contact_urn(contact_urn, uuid_table)
+            contact_urn = _normalise_and_validate_contact_urn(contact.urns[0])
+
+            if rapid_pro_config.uuid_filter is not None:
+                # If a uuid filter exists, then only add this message if the sender's uuid exists in the uuid table
+                # and in the valid uuids. The check for presence in the uuid table is to ensure we don't add a uuid
+                # table entry for people who didn't consent for us to continue to keep their data.
+                if not uuid_table.has_data(contact_urn) or uuid_table.data_to_uuid(contact_urn) not in valid_participant_uuids:
```

Comment:
### 
```suggestion
                if not uuid_table.has_data(contact_urn) and uuid_table.data_to_uuid(contact_urn) not in valid_participant_uuids:
```

Reply:
hmm if we make this an 'and' then if the uuid_table doesn't have the phone number it will de-identify it to check that its in the valid_participant_uuids list, which means we copy phone numbers from people who didn't opt-in to the pool into our new tables. 

There's also no way that a contact can be in the valid_participant_uuids without having been de-identified first...